File: backend/Helpers/BusinessSettings.cs
namespace Auto_Wash.Helpers
{
    /// <summary>
    /// Centralized single source of truth for business rules, demo scaling, tier multipliers, and voucher discount caps.
    /// Eliminates magic numbers across the backend.
    /// </summary>
    public static class BusinessSettings
    {
        // Pricing Scale for Demo Mode (Original price * 0.1)
        public const decimal DemoPriceScaleFactor = 0.1m;

        // Default Loyalty Config (1000 VND = 10 Points in Demo Mode)
        public const int DemoPointsPerThousandVND = 10;
        public const int BusinessPointsPerThousandVND = 1;

        // Tier Multipliers
        public const decimal TierMultiplierMember = 1.00m;
        public const decimal TierMultiplierSilver = 1.25m;
        public const decimal TierMultiplierGold = 1.50m;
        public const decimal TierMultiplierPlatinum = 2.00m;

        // Maximum Voucher Discount Limits (VND)
        public const int MaxDiscount5Percent = 5000;
        public const int MaxDiscount10Percent = 10000;
        public const int MaxDiscount15Percent = 15000;

        // Upgraded Tier Spending Thresholds (Demo Mode Scale)
        public const int TierThresholdMemberMin = 0;
        public const int TierThresholdSilverMin = 50000;
        public const int TierThresholdSilverMaintain = 30000;
        public const int TierThresholdGoldMin = 150000;
        public const int TierThresholdGoldMaintain = 100000;
        public const int TierThresholdPlatinumMin = 300000;
        public const int TierThresholdPlatinumMaintain = 200000;

        /// <summary>
        /// Calculates the maximum allowed discount for a percentage-based voucher.
        /// </summary>
        public static int GetMaxDiscountCap(decimal discountPercent)
        {
            if (discountPercent <= 5.0m) return MaxDiscount5Percent;
            if (discountPercent <= 10.0m) return MaxDiscount10Percent;
            return MaxDiscount15Percent;
        }
    }
}

File: frontend/src/constants/businessSettings.js
/**
 * Centralized Business Settings for AutoWash Pro Frontend.
 * Eliminates magic numbers and inline business rules across React components.
 */

export const BUSINESS_SETTINGS = {
  // Demo Mode Price Scale Factor
  DEMO_SCALE_FACTOR: 0.1,

  // Service Base Prices (Demo Scale)
  SERVICE_PRICES: {
    STANDARD_WASH: 14900,
    PREMIUM_WASH: 29900,
    WAX_COATING: 7900,
    NANO_CERAMIC: 19900,
    ENGINE_BAY: 9900,
    ODOR_REMOVAL: 6900,
    LEATHER_SEAT: 12900,
    HEADLIGHT_RESTORATION: 15900,
  },

  // Tier Multipliers & Perks
  TIERS: {
    MEMBER: {
      name: 'Member',
      multiplier: 1.0,
      discountPercent: 0,
      minSpend: 0,
      badgeColor: '#6B7280',
      badgeClass: 'badge-member',
    },
    SILVER: {
      name: 'Silver',
      multiplier: 1.25,
      discountPercent: 2,
      minSpend: 50000,
      badgeColor: '#9CA3AF',
      badgeClass: 'badge-silver',
    },
    GOLD: {
      name: 'Gold',
      multiplier: 1.5,
      discountPercent: 5,
      minSpend: 150000,
      badgeColor: '#F59E0B',
      badgeClass: 'badge-gold',
    },
    PLATINUM: {
      name: 'Platinum',
      multiplier: 2.0,
      discountPercent: 10,
      minSpend: 300000,
      badgeColor: '#8B5CF6',
      badgeClass: 'badge-platinum',
    },
  },

  // Points Redemption Config
  LOYALTY: {
    POINTS_PER_1000_VND_DEMO: 10,
    POINTS_PER_1000_VND_BUSINESS: 1,
    MONEY_REDEMPTIONS: [
      { points: 500, value: 1000, label: 'Voucher 1.000đ' },
      { points: 1000, value: 2000, label: 'Voucher 2.000đ' },
      { points: 2500, value: 5000, label: 'Voucher 5.000đ' },
      { points: 5000, value: 10000, label: 'Voucher 10.000đ' },
    ],
    PERCENT_REDEMPTIONS: [
      { points: 3500, percent: 5, maxDiscount: 5000, label: 'Giảm giá 5% (Tối đa 5.000đ)' },
      { points: 7000, percent: 10, maxDiscount: 10000, label: 'Giảm giá 10% (Tối đa 10.000đ)' },
    ],
  },

  // Maximum Voucher Discount Limits
  MAX_DISCOUNT_CAPS: {
    PERCENT_5: 5000,
    PERCENT_10: 10000,
    PERCENT_15: 15000,
  },
};

/**
 * Currency Formatter Utility
 */
export const formatVND = (amount) => {
  if (amount === null || amount === undefined || isNaN(amount)) return '0đ';
  return new Intl.NumberFormat('vi-VN', { style: 'currency', currency: 'VND' })
    .format(amount)
    .replace('₫', 'đ');
};
